pkg/tests/end_to_end_tests/trace_tree_test.go

```diff
@@ -0,0 +1,392 @@
+		from downstream_spans('3dadb2bf-0035-433e-b74b-9075cc9260e8', 2) 
+		order by span_id`
+		expectedResults := []result{
+			{2, 1, 0, 1, "{2}"},
+			{5, 2, 1, 2, "{2,5}"},
+			{6, 2, 1, 2, "{2,6}"},
+			{7, 2, 1, 2, "{2,7}"},
+			{13, 6, 2, 3, "{2,6,13}"},
+			{16, 13, 3, 4, "{2,6,13,16}"},
+			{17, 13, 3, 4, "{2,6,13,17}"},
+		}
+		eval(t, qry, expectedResults)
+	})
+
+	t.Run("downstream_spans_max_dist", func(t *testing.T) {
+		const qry = `
+		select span_id, coalesce(parent_span_id, -1), dist, idx, path::text
+		from downstream_spans('3dadb2bf-0035-433e-b74b-9075cc9260e8', 2, 2) 
+		order by span_id`
+		expectedResults := []result{
+			{2, 1, 0, 1, "{2}"},
+			{5, 2, 1, 2, "{2,5}"},
+			{6, 2, 1, 2, "{2,6}"},
+			{7, 2, 1, 2, "{2,7}"},
+			{13, 6, 2, 3, "{2,6,13}"},
+		}
+		eval(t, qry, expectedResults)
+	})
+}
+
+func testSiblingSpans(t *testing.T, ctx context.Context, db *pgxpool.Pool) {
+	const qry = `
+	select span_id, coalesce(parent_span_id, -1)
+	from sibling_spans('3dadb2bf-0035-433e-b74b-9075cc9260e8', 8) 
+	order by span_id`
+
+	type result struct {
+		SpanId       int64
+		ParentSpanId int64
+	}
+
+	t.Run("sibling_spans", func(t *testing.T) {
+		expectedResults := []result{
+			{8, 3},
+			{9, 3},
+			{10, 3},
+		}
+		rows, err := db.Query(ctx, qry)
+		require.NoError(t, err, "Error running sibling_spans: %v", err)
+		defer rows.Close()
+
+		for _, expected := range expectedResults {
+			if !rows.Next() {
+				t.Fatalf("Expected a row for span id %v but got none", expected.SpanId)
+			}
+
+			var actual = result{}
+			err = rows.Scan(&actual.SpanId, &actual.ParentSpanId)
+			require.NoError(t, err, "Failed to scan row. Expected row for span id %v: %v", expected.SpanId, err)
+
+			require.Equal(t, expected, actual)
+		}
+	})
+}
+
+func testSpanTree(t *testing.T, ctx context.Context, db *pgxpool.Pool) {
+	type result struct {
+		SpanId       int64
+		ParentSpanId int64
+		Dist         int64
+		Index        int64
+		Path         string
+	}
+
+	eval := func(t *testing.T, qry string, expectedResults []result) {
+		rows, err := db.Query(ctx, qry)
+		require.NoError(t, err, "Error running span_tree: %v", err)
+		defer rows.Close()
+
+		for _, expected := range expectedResults {
+			if !rows.Next() {
+				t.Fatalf("Expected a row for span id %v but got none", expected.SpanId)
+			}
+
+			var actual = result{}
+			err = rows.Scan(&actual.SpanId, &actual.ParentSpanId, &actual.Dist, &actual.Index, &actual.Path)
+			require.NoError(t, err, "Failed to scan row. Expected row for span id %v: %v", expected.SpanId, err)
+
+			require.Equal(t, expected, actual)
+		}
+		require.False(t, rows.Next(), "Found more result rows than expected")
+	}
+
+	t.Run("span_tree", func(t *testing.T) {
+		const qry = `
+		select span_id, coalesce(parent_span_id, -1), dist, idx, path::text
+		from span_tree('3dadb2bf-0035-433e-b74b-9075cc9260e8', 6) 
+		order by span_id`
+		expectedResults := []result{
+			{1, -1, -2, 3, "{6,2,1}"},
```

Comment:
same question here about ordering of "{6,2,1}" vs "{1,2,6}"

Reply:
Same as above. It's rendered from the "perspective" of the span_id passed as an arg.

Reply:
changed this. paths are always ordered from upstream to downstream for consistency.